libs/orders/src/services/fulfillment/create-fulfillment/service.ts

```diff
@@ -0,0 +1,61 @@
+import { getPrismaClient } from '@vestido-ecommerce/models';
+
+import { CreateFulfillmentSchema, CreateFulfillmentSchemaType } from './zod';
+
+export async function createFulfillment(data: CreateFulfillmentSchemaType) {
+  const prisma = getPrismaClient();
+
+  // Validate the input data
+  const validatedData = CreateFulfillmentSchema.parse(data);
+
+  // Execute the operations within a transaction
+  const result = await prisma.$transaction(async (prisma) => {
+    // Create the fulfillment
+    const newFulfillment = await prisma.fulfillment.create({
+      data: {
+        orderId: validatedData.orderId,
+        status: 'DRAFT',
+        length: validatedData.length,
+        breadth: validatedData.breadth,
+        height: validatedData.height,
+        weight: validatedData.weight,
+        fulfillmentItems: {
+          createMany: {
+            data: validatedData.items.map((i) => ({
+              orderItemId: i.orderItemId,
+              quantity: i.quantity,
+            })),
+          },
+        },
+      },
+    });
+
+    // Extract orderItemIds from the fulfillmentItems
+    const orderItemIds = validatedData.items.map((item) => item.orderItemId);
```

Comment:
How can we be sure that all the orderItems passed belongs to the Order? 😄

Reply:
Now only fulfillment table and flfillmentItem table is created in this service and nothing else is updated.